packages/vm/src/modules/pinned_memory_cache.rs

```diff
@@ -40,6 +41,19 @@ impl PinnedMemoryCache {
     pub fn has(&self, checksum: &Checksum) -> bool {
         self.modules.contains_key(checksum)
     }
+
+    /// Returns the number of elements in the cache.
+    pub fn len(&self) -> usize {
+        self.modules.len()
+    }
+
+    /// Returns cumulative size of all elements in the cache.
+    ///
+    /// This is based on the values provided with `store`. No actual
+    /// memory size is measured here.
+    pub fn size(&self) -> usize {
+        self.modules.iter().map(|(_, module)| module.size).sum()
```

Comment:
This is not a constant operation. Depending on how often these stats will be gathered, we might want to improve on this. Basically, by keeping a running total of module sizes in the map.

Reply:
Yeah, good point. Thought about it as well but then realized that we will only have very few contracts pinned. If that changes, we can improve the implementation without breaking interfaces.